server/src-lib/Hasura/Eventing/EventTrigger.hs

```diff
@@ -408,10 +408,11 @@ fetchEvents :: Int -> Q.TxE QErr [Event]
 fetchEvents limitI =
   map uncurryEvent <$> Q.listQE defaultTxErrorHandler [Q.sql|
       UPDATE hdb_catalog.event_log
-      SET locked = 't'
+      SET locked = NOW()
       WHERE id IN ( SELECT l.id
                     FROM hdb_catalog.event_log l
-                    WHERE l.delivered = 'f' and l.error = 'f' and l.locked = 'f'
+                    WHERE l.delivered = 'f' and l.error = 'f' 
+                          and (l.locked IS NULL or l.locked < NOW() - interval '30 minute')
```

Comment:
```suggestion
                          and (l.locked IS NULL or l.locked < (NOW() - interval '30 minute'))
```

as a very non-expert on what implicit type conversions might be happening in places in postgres, this makes me more comfortable